web/src/lib/managers/photostream-manager/PhotostreamSegment.svelte.ts

```diff
@@ -0,0 +1,150 @@
+import { CancellableTask } from '$lib/utils/cancellable-task';
+import { handleError } from '$lib/utils/handle-error';
+import { t } from 'svelte-i18n';
+import { get } from 'svelte/store';
+
+import type { PhotostreamManager } from '$lib/managers/photostream-manager/PhotostreamManager.svelte';
+import type { TimelineAsset } from '$lib/managers/timeline-manager/types';
+import type { ViewerAsset } from '$lib/managers/timeline-manager/viewer-asset.svelte';
+
+export type SegmentIdentifier = {
+  matches(segment: PhotostreamSegment): boolean;
+};
+export abstract class PhotostreamSegment {
+  #intersecting = $state(false);
+  actuallyIntersecting = $state(false);
+  #isLoaded = $state(false);
+
+  #height = $state(0);
+  #top = $state(0);
+  #assets = $derived.by(() => this.viewerAssets.map((viewerAsset) => viewerAsset.asset));
+
+  initialCount = $state(0);
+  percent = $state(0);
+
+  assetsCount = $derived.by(() => (this.isLoaded ? this.viewerAssets.length : this.initialCount));
+  loader = new CancellableTask(
+    () => this.markLoaded(),
+    () => this.markCanceled,
+    () => this.handleLoadError,
+  );
+  isHeightActual = $state(false);
+
+  abstract get timelineManager(): PhotostreamManager;
```

Comment:
If this still uses a timeline manager, the class should be called a TimelineSegment